mlrun/datastore/sources.py

```diff
@@ -950,12 +1026,18 @@ def to_dataframe(self):
         if table_name and db_path:
             engine = db.create_engine(db_path)
             with engine.connect() as con:
-                return pd.read_sql(
-                    query,
-                    con=con,
-                    params=params,
-                    chunksize=self.attributes.get("chunksize"),
-                    parse_dates=self.attributes.get("time_fields"),
+                return filter_df_start_end_time(
+                    pd.read_sql(
+                        query,
+                        con=con,
+                        params=params,
+                        chunksize=self.attributes.get("chunksize"),
+                        parse_dates=self.attributes.get("time_fields"),
+                        columns=columns,
```

Comment:
This is one place where you're first filtering columns and then filtering based on time. This will fail in a corner case where the `time_field` is not in the `columns` and yet you ask to time-filter based on it. In all other places you first do time filtering so it will work, but here it's reversed.